### PR TITLE
Remove IsStandardContract

### PR DESCRIPTION
**Pull request by shargon**
Close https://github.com/neo-project/neo/issues/2399

### CODE REVIEW

File: src/neo/SmartContract/ApplicationEngine.Contract.cs [hunk 1]
```diff
@@ -62,7 +62,7 @@ protected internal bool IsStandardContract(UInt160 hash)
 
             // Try to find it in the transaction
 
-            if (ScriptContainer is Transaction tx)
+            if (ScriptContainer is Transaction tx && tx.Witnesses is not null)
```
Comment (csmuller) [inline src/neo/SmartContract/ApplicationEngine.Contract.cs:65]: Will this work for the case of `invokefunction` RPC method calls?
I.e., if I call a contract method with `invokefunction` and the method contains a call to `System.Contract.isStandard`.

Comment (shargon) [reply src/neo/SmartContract/ApplicationEngine.Contract.cs:65]: Yes, it should return false if there are no witness

**Comment by cloud8little** [comment]
Test when invokefunction with wallet open, since the wallet not sign, witness is null.

cacdc5861aa7c74323ccba31dfcec40

then send the transaction, check applicationlog, this time witness is not null.
e80fee4fe8cc0ea8fcf4b80c3ff12cf

**Review by cloud8little — DISMISSED** [review]

**Comment by superboyiii** [comment]
@erikzhang Need your review.

**Comment by erikzhang** [comment]
I think we can just remove the syscall `IsStandardContract`. It is useless now.

**Comment by superboyiii** [comment]
@shargon Do we have any scenario to call `IsStandardContract`? Seems it's impossible to do this.

**Comment by shargon** [comment]
> @shargon Do we have any scenario to call IsStandardContract? Seems it's impossible to do this in contract.

No, @roman-khimov what do you think?

**Comment by roman-khimov** [comment]
> > @shargon Do we have any scenario to call IsStandardContract? Seems it's impossible to do this in contract.
> 
> No, @roman-khimov what do you think?

#1753, option №4 wins.

**Review by erikzhang — APPROVED** [review]

**Comment by superboyiii** [comment]
Merge?

**Comment by dusmart** [comment]
Please remove this syscall's price in document. Thanks.
https://docs.neo.org/docs/zh-cn/reference/fees.html.

**Pull request merged**